teachertool/src/components/CriteriaInstanceDisplay.tsx

```diff
@@ -0,0 +1,109 @@
+import { classList } from "react-common/components/util";
+import { splitCriteriaTemplate } from "../utils";
+// eslint-disable-next-line import/no-internal-modules
+import css from "./styling/CriteriaInstanceDisplay.module.scss";
+
+interface InlineInputSegmentProps {
+    initialValue: string;
+    instance: CriteriaInstance;
+    param: CriteriaParameterValue;
+    shouldExpand: boolean;
+    numeric: boolean;
+}
+const InlineInputSegment: React.FC<InlineInputSegmentProps> = ({
+    initialValue,
+    instance,
+    param,
+    shouldExpand,
+    numeric,
+}) => {
+    function onChange(newValue: string) {
+        setParameterValue(instance.instanceId, param.name, newValue);
+    }
+
+    return (
+        <DebouncedInput
+            className={classList(
+                css["inline-input"],
+                numeric ? css["number-input"] : css["string-input"],
+                shouldExpand ? css["long"] : undefined,
+            )}
+            initialValue={initialValue}
+            onChange={onChange}
+            preserveValueOnBlur={true}
+            placeholder={numeric ? "0" : param.name}
+            title={param.name}
+            autoComplete={false}
+            type = {numeric ? "number" : "text"}
+        />
+    );
+};
+
+interface CriteriaInstanceDisplayProps {
+    criteriaInstance: CriteriaInstance;
+}
+
+export const CriteriaInstanceDisplay: React.FC<CriteriaInstanceDisplayProps> = ({ criteriaInstance }) => {
+    const catalogCriteria = getCatalogCriteriaWithId(criteriaInstance.catalogCriteriaId);
+    if (!catalogCriteria) {
+        return null;
+    }
+
+    function getParameterSegmentDisplay(paramName: string): JSX.Element | null {
+        if (!paramName) {
+            return null;
+        }
+        paramName = paramName.toLocaleLowerCase();
+
+        const paramDef = catalogCriteria?.params?.find(p => p.name === paramName);
+        const paramInstance = criteriaInstance?.params?.find(p => p.name === paramName);
+        if (!paramDef || !paramInstance) {
+            logDebug(`Missing info for '${paramName}': paramDef=${paramDef}, paramInstance=${paramInstance}`);
+            return null;
+        }
+
+        switch (paramDef.type) {
+            case "string":
+            case "number":
+                return (
+                    <InlineInputSegment
+                        initialValue={paramInstance.value}
+                        param={paramInstance}
+                        instance={criteriaInstance}
+                        shouldExpand={paramDef.picker === "longString"}
+                        numeric={paramDef.type === "number"}
+                    />
+                );
+            case "block": // TODO
+            default:
+                return null;
+        }
+    }
+
+    function getPlainTextSegmentDisplay(text: string): JSX.Element | null {
+        return text ? <div className={css["text-segment"]}>{text}</div> : null;
```

Comment:
I think it's useful so we don't create an empty div when the text is an empty string, and in this case, it felt slightly tidier to do that check here rather than before calling this function.